crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -483,21 +483,33 @@ impl SlidingSync {
             if self.inner.sticky.read().unwrap().data().extensions.e2ee.enabled == Some(true) {
                 debug!("Sliding Sync is sending the request along with outgoing E2EE requests");
 
-                let (e2ee_uploads, response) =
-                    futures_util::future::join(self.inner.client.send_outgoing_requests(), request)
-                        .await;
-
-                if let Err(error) = e2ee_uploads {
+                // If the sliding sync request fails faster than the e2ee requests get their
+                // responses, we want to fail as early as possible, so as to invalidate the
+                // sliding sync session as early as possible.
+                //
+                // Also, we don't want an interruption in the sliding sync request to abort
+                // processing of the e2ee response.
+                //
+                // For those reasons, start the e2ee requests in a background task.
+
+                let client = self.inner.client.clone();
+                let e2ee_uploads = spawn(async move { client.send_outgoing_requests().await });
```

Comment:
Do we want a `TryMerge` here (cf https://docs.rs/futures-concurrency/latest/futures_concurrency/future/index.html#fallible-concurrency)? Sadly, it doesn't seem implemented in `futures_concurrency` yet.

I need to review this clearly, and see how it can impact us. The task isn't detached, so the execution order is guaranteed. It means we cannot have multiple tasks running in different order here.

It looks OK to me. I don't see anything problematic here.

Reply:
Well it becomes detached in the case the SS request returns an error, no?

Reply:
🤦  of course yeah!

Is it something we want? Can't we have a problem with this? I'm not comfortable having a detached task.

Reply:
It likely won't be problematic, though we do introduce a race where we could call `send_outgoing_requests()` before the detached task is done.

This would lead sending the same requests out twice, not the end of the world I guess since most of them will be either deduplicated by the server or become a noop. We likely should protect against this with a lock inside `send_outgoing_requests()`.